src/sentry/services/hybrid_cloud/__init__.py

```diff
@@ -0,0 +1,322 @@
+        an appropriate set.
+        :param user_id:
+        When null, this should imply the entire set of organizations, not bound by user.  Be certain to authenticate
+        users before returning this.
+        :param scope:
+        :param only_visible:
+        :return:
+        """
+        pass
+
+    @abstractmethod
+    def get_organization_by_slug(
+        self, slug: str, only_visible: bool, allow_stale: bool
+    ) -> Optional[ApiOrganization]:
+        pass
+
+    def _serialize_organization(self, org: Organization) -> ApiOrganization:
+        return ApiOrganization(
+            slug=org.slug,
+            id=org.id,
+        )
+
+
+class DatabaseBackedOrganizationService(OrganizationService):
+    def get_organization_by_slug(
+        self, slug: str, only_visible: bool, allow_stale: bool
+    ) -> Optional[ApiOrganization]:
+        try:
+            if allow_stale:
+                org = Organization.objects.get_from_cache(slug=slug)
+            else:
+                org = Organization.objects.get(slug=slug)
+
+            if only_visible and org.status != OrganizationStatus.VISIBLE:
+                raise Organization.DoesNotExist
+            return self._serialize_organization(org)
+        except Organization.DoesNotExist:
+            logger.info("Active organization [%s] not found", slug)
+
+        return None
+
+    def close(self):
+        pass
+
+    def get_organizations(
+        self, user_id: Optional[id], scope: Optional[str], only_visible: bool
+    ) -> List[ApiOrganization]:
+        from django.conf import settings
+
+        if settings.SENTRY_PUBLIC and scope is None:
+            if only_visible:
+                return list(Organization.objects.filter(status=OrganizationStatus.ACTIVE))
+            else:
+                return list(Organization.objects.filter())
+
+        if user_id is not None:
+            qs = OrganizationMember.objects.filter(user_id=user_id)
+        else:
+            qs = OrganizationMember.objects.filter()
+
+        qs = qs.select_related("organization")
+        if only_visible:
+            qs = qs.filter(organization__status=OrganizationStatus.ACTIVE)
+
+        results = list(qs)
+
+        if scope is not None:
+            return [r.organization for r in results if scope in r.get_scopes()]
+
+        return [self._serialize_organization(r.organization) for r in results]
+
+
+class DatabaseBackedProjectKeyService(ProjectKeyService):
+    def close(self):
+        pass
+
+    def get_project_key(self, project_id: str, role: ProjectKeyRole) -> Optional[ApiProjectKey]:
+        from sentry.models import ProjectKey
+
+        project_keys = ProjectKey.objects.filter(
+            project=project_id, roles=F("roles").bitor(role.as_orm_role())
+        )
+
+        if project_keys:
+            return ApiProjectKey(dsn_public=project_keys[0].dsn_public)
+
+        return None
+
+
+def CreateStubFromBase(base: Type[ServiceInterface]) -> Type[ServiceInterface]:
```

Comment:
Good point.  I don't know how common it is, because I tend to think of upper letter functions or variables as being type-order or kind-ordered logic.  So Super => a type, Con =>  a type.  But it may be the case that in the wild these things are `cls` or something.